fastestimator/summary/system.py

```diff
@@ -74,17 +75,43 @@ def __init__(self,
 
         self.network = network
         self.mode = mode
-        self.global_step = None
         self.num_devices = num_devices
         self.log_steps = log_steps
         self.total_epochs = total_epochs
-        self.epoch_idx = None
         self.batch_idx = None
         self.max_train_steps_per_epoch = max_train_steps_per_epoch
         self.max_eval_steps_per_epoch = max_eval_steps_per_epoch
         self.stop_training = False
         self.summary = Summary(None)
         self.experiment_time = ""
+        self._initialize_state()
+
+    def _initialize_state(self) -> None:
+        """Initialize the training state.
+        """
+        self.global_step = None
+        self.epoch_idx = 0
```

Comment:
I think we have code somewhere looking for if epoch_idx is None. Need to double check

Reply:
checked it, didn't find any